### PR TITLE
Return ROS time values

### PR DESCRIPTION
**Pull request by liamembark**
Fixes: DATA-627

Description of Changes
======================
This switches the python interface to providing timestamps as ROS times, similar to how the actual rospy interface works.

Test Plan
=========
Unit tests.

### CODE REVIEW

File: python/adapters.h [hunk 1]
```diff
@@ -21,21 +21,21 @@ typedef std::unordered_set<Embag::RosValue::Type, EnumClassHash> RosValueTypeSet
...
+py::object castValue(const Embag::RosValue::Pointer &value, py::object ros_time_py_type);
 
 // By default, it doesn't make sense to return a memoryview for python non-primitive types
 const RosValueTypeSet default_types_to_unpack = {
   Embag::RosValue::Type::ros_duration,
   Embag::RosValue::Type::ros_time
 };
 
-py::object primitiveArrayToPyObject(const Embag::RosValue::Pointer &primitive_array, const RosValueTypeSet &types_to_unpack=default_types_to_unpack) {
+py::object primitiveArrayToPyObject(const Embag::RosValue::Pointer &primitive_array, const RosValueTypeSet &types_to_unpack=default_types_to_unpack, py::object ros_time_py_type=py::none()) {
```
Comment (kashish-jain) [inline python/adapters.h:34]: NIT: These lines seem to be too big. Can we re-format ?

File: test/embag_test_python.py [hunk 2]
```diff
@@ -201,5 +201,12 @@ def testDictUnpacking(self):
             assert isinstance(msg.dict()['pose']['covariance'], memoryview if sys.version_info >= (3,3) else np.ndarray)
             assert isinstance(msg.dict(types_to_unpack={embag.RosValueType.float64})['pose']['covariance'], list)
 
+    def testROSTimeDicting(self):
+        for msg in self.view.getMessages('/base_pose_ground_truth'):
+            assert isinstance(msg.dict()['header']['stamp'], embag.RosTime)
+            assert isinstance(msg.dict(ros_time_py_type=None)['header']['stamp'], embag.RosTime)
```
Comment (kashish-jain) [inline test/embag_test_python.py:207]: Can we also test `to_sec` and `to_nsec` just to make sure that casting is working fine

**Review by kashish-jain — COMMENTED** [review]
I'm wondering if we need to support getting time as `int` or `float`? Like, why can't we just return `embag.RosTime` and the client can call `to_sec` or `to_nsec` to get long/int or float. This way we won't have to pass around that `ros_time_py_type`. Does rospy support just getting time/duration as primitive? May be this is for backward compatibility?

**Comment by liamembark** [comment]
> I'm wondering if we need to support getting time as `int` or `float`? Like, why can't we just return `embag.RosTime` and the client can call `to_sec` or `to_nsec` to get long/int or float. This way we won't have to pass around that `ros_time_py_type`. Does rospy support just getting time/duration as primitive? May be this is for backward compatibility?

@kashish-jain Its mostly just about providing nice functionality for clients so that they don't have to worry about writing the conversions/serializations themselves and dealing with performance hits from deeply nested structures. IMO passing the arg around isn't too big of a deal.

rospy doesn't provide any to_dict functionality, so theres no precedence there.

File: python/adapters.h (hunk 1 above)
Comment (liamembark) [reply python/adapters.h:34]: Sure thing

File: test/embag_test_python.py (hunk 2 above)
Comment (kashish-jain) [reply test/embag_test_python.py:207]: May be have something like 

```
x = msg.dict(ros_time_py_type=None)['header']['stamp']
assert x.to_sec() == x.to_nsec()*(10**9) == msg.dict(ros_time_py_type=int)['header']['stamp'] == msg.dict(ros_time_py_type=float)['header']['stamp'] * 10**9
```

Comment (liamembark) [reply test/embag_test_python.py:207]: I like it!

**Review by kashish-jain — APPROVED** [review]
🚢

**Review by brendangeck — APPROVED** [review]

**Pull request merged**